### ISSUE TITLE
Input field in nested routes (go_router) lose focus and functionality when navigated back from a child route using swipe gesture instead of system(back-button/swipe). 

### ISSUE THREAD

**Issue by ignicubeLaeeq** [flutter/flutter#151582]
### What package does this bug report belong to?

go_router

### What target platforms are you seeing this bug on?

Android

### Have you already upgraded your packages?

Yes

### Dependency versions

<details><summary>pubspec.lock</summary>

```lock
  go_router:
    dependency: "direct main"
    description:
      name: go_router
      sha256: b465e99ce64ba75e61c8c0ce3d87b66d8ac07f0b35d0a7e0263fcfc10f99e836
      url: "https://pub.dev"
    source: hosted
    version: "13.2.5"
```

</details>

### Steps to reproduce

1. Add nested routes with TextFormFields.
2. Navigate to bottom most route with context.goName function. 
3. Navigate back with swipe gesture. (Works fine with system back button). 

### Expected results

The textfield should be focusable and editable. 

### Actual results

Textfield cannot be edited. 

### Code sample

<details open><summary>Code sample</summary>

```dart
        GoRoute(
          name: AuthScreen.route.name,
          path: AuthScreen.route.path,
          builder: (context, state) {
            return const AuthScreen();
          },
          routes: [
            GoRoute(
                name: EmailScreen.route.name,
                path: EmailScreen.route.path,
                builder: (context, state) {
                  return const EmailScreen();
                },
                routes: [
                  GoRoute(
                    name: SignupScreen.route.name,
                    path: SignupScreen.route.path,
                    builder: (context, state) {
                      final extra = state.extra as Map<String, dynamic>;
                      final email = extra['email'] as String;
                      return SignupScreen(email: email);
                    },
                  ),
                  GoRoute(
                    name: PasswordScreen.route.name,
                    path: PasswordScreen.route.path,
                    builder: (context, state) {
                      final extra = state.extra as Map<String, dynamic>;
                      final email = extra['email'] as String;
                      final name = extra['name'] as String?;
                      return PasswordScreen(email: email, name: name);
                    },
                  ),
                ]),
          ],
        ),
```

</details>

### Screenshots or Videos

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @ignicubeLaeeq, you are using an older version of go_router. Kindly upgrade to the latest version of go_router to see if you still experience this issue. If you do, please provide the new version you're using along with a complete minimal reproducible code sample so that we can properly investigate this issue. 

Please also provide the output of `flutter doctor -v`. 

Thank you

**Comment by ignicubeLaeeq** [comment]
Hi @danagbemava-nc 
Thank you for your response. 
I have updated the package which had no effect. It turns out the issue does not originate from go_router. 
Instead following lines in my themeData appear to be causing the problem. 

```
  pageTransitionsTheme: const PageTransitionsTheme(
    builders: {
      TargetPlatform.android: CupertinoPageTransitionsBuilder(),
      TargetPlatform.iOS: CupertinoPageTransitionsBuilder(),
    },
  ),

```
After removing these lines i can no longer swipe to go back to previous route and only rely on system's navigation buttons.

**Comment by ignicubeLaeeq** [comment]
Complete example:
```
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';

void main() {
  final GlobalKey<NavigatorState> navigatorKey =
      GlobalKey<NavigatorState>(debugLabel: 'navigator');

  final GoRouter goRouter = GoRouter(
    navigatorKey: navigatorKey,
    initialLocation: '/',
    routes: [
      GoRoute(
        path: '/',
        name: '/',
        builder: (context, state) => const AuthScreen(),
      ),
      GoRoute(
        path: '/routeA',
        name: 'routeA',
        builder: (context, state) => const AScreen(),
        routes: [
          GoRoute(
            path: 'routeB',
            name: 'routeB',
            builder: (context, state) => const BScreen(),
            routes: [
              GoRoute(
                path: 'routeC',
                name: 'routeC',
                builder: (context, state) => const CScreen(),
              ),
            ],
          ),
        ],
      ),
    ],
  );
  runApp(MaterialApp.router(
    title: 'Flutter Demo',
    theme: ThemeData(
      colorScheme: ColorScheme.fromSeed(seedColor: Colors.red),
      useMaterial3: true,
      pageTransitionsTheme: const PageTransitionsTheme(
        builders: {
          TargetPlatform.android: CupertinoPageTransitionsBuilder(),
          TargetPlatform.iOS: CupertinoPageTransitionsBuilder(),
        },
      ),
    ),
    debugShowCheckedModeBanner: false,
    routerConfig: goRouter,
  ));
}

class AScreen extends StatelessWidget {
  const AScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: Scaffold(
        appBar: AppBar(
          title: const Text("A Screen"),
        ),
        body: SingleChildScrollView(
          child: Column(
            children: [
              TextFormField(
                decoration: const InputDecoration(hintText: 'Test field'),
              ),
              OutlinedButton(
                  onPressed: () {
                    context.pushNamed('routeB');
                  },
                  child: const Text('Go to route B'))
            ],
          ),
        ),
      ),
    );
  }
}

class BScreen extends StatelessWidget {
  const BScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: Scaffold(
        appBar: AppBar(
          title: const Text("B Screen"),
        ),
        body: SingleChildScrollView(
          child: Column(
            children: [
              TextFormField(
                decoration: const InputDecoration(hintText: 'Test field'),
              ),
              OutlinedButton(
                  onPressed: () {
                    context.pushNamed('routeC');
                  },
                  child: const Text('Go to route C'))
            ],
          ),
        ),
      ),
    );
  }
}

class CScreen extends StatelessWidget {
  const CScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: Scaffold(
        appBar: AppBar(
          title: const Text("C Screen"),
        ),
        body: const SingleChildScrollView(
          child: Column(
            children: [
              Text('Swipe right to go back'),
            ],
          ),
        ),
      ),
    );
  }
}

class AuthScreen extends StatelessWidget {
  const AuthScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: OutlinedButton(
          child: const Text('Login'),
          onPressed: () {
            context.pushNamed('routeA');
          },
        ),
      ),
    );
  }
}

```

**Comment by ignicubeLaeeq** [comment]
flutter doctor -v
```
[✓] Flutter (Channel stable, 3.22.1, on Ubuntu 22.04.4 LTS 6.5.0-41-generic, locale en_US.UTF-8)
    • Flutter version 3.22.1 on channel stable at /home/laeeq/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a14f74ff3a (7 weeks ago), 2024-05-22 11:08:21 -0500
    • Engine revision 55eae6864b
    • Dart version 3.4.1
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /home/laeeq/Android/Sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /home/laeeq/Android/Sdk
    • Java binary at: /home/laeeq/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2023.1)
    • Android Studio at /home/laeeq/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.90.0)
    • VS Code at /usr/share/code
    • Flutter extension version 3.92.0

[✓] Connected device (3 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 14 (API 34) (emulator)
    • Linux (desktop)              • linux         • linux-x64      • Ubuntu 22.04.4 LTS 6.5.0-41-generic
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 122.0.6261.111

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

**Comment by danagbemava-nc** [comment]
Hi @ignicubeLaeeq, what device are you seeing this issue on? It seems to work fine on my Pixel 7. See the recording below. Can you share a recording of what you're seeing? 


https://github.com/user-attachments/assets/6fcb504c-a2d3-4266-9dd2-7506766956d1

**Comment by ignicubeLaeeq** [comment]
@danagbemava-nc 
Steps to produce: 
1. Enable system navigation buttons (this will disable system swipe to go back)
2. Run the app
3. Go to A screen
4. Type something in inside the textfield. 
5. Click the button to B Screen
6. Swipe right to go back to A screen. 
7. Input field is disabled, keypad doesnt pop up. 

Note: Navigation with system back button and back button in the app bar works fine. 
The problem occurs only with the swipe introduced by CupertinoPageTransitionsBuilder(), if i remove this transition from the theme data, the swipe will have no effect. 
[Screencast from 12-07-2024 13:55:01.webm](https://github.com/user-attachments/assets/1aa8b574-730c-45d5-b4f7-3bffc067b4b7)

**Comment by danagbemava-nc** [comment]
Hi @ignicubeLaeeq, I just tried again with the steps provided above and I still cannot reproduce the issue on my pixel 7. See the recording below: 

PS: I'm not using the gesture-based navigation and my screen recorder doesn't show the taps on screen. 

Can you try on a physical device to see if you still experience the issue? 

https://github.com/user-attachments/assets/313efd95-58e1-40cd-9b48-02e24eb07fd0

**Comment by Menusyes** [comment]
I also encountered this problem, the same happened in the  swipe gesture come back of `CupertinoPageTransitionsBuilder`.

**Comment by Menusyes** [comment]
> Hi @ignicubeLaeeq, I just tried again with the steps provided above and I still cannot reproduce the issue on my pixel 7. See the recording below:
> 
> PS: I'm not using the gesture-based navigation and my screen recorder doesn't show the taps on screen.
> 
> Can you try on a physical device to see if you still experience the issue?
> 
>  screen-20240715-064641.mp4


I reproduced this problem. Although it was in the PC Android emulator, the behavior was the same on the real machine.

![GIF 2024-7-15 15-01-39](https://github.com/user-attachments/assets/83005749-7fc9-4a2a-af7a-c67baf5b7103)

**Comment by Menusyes** [comment]
After verification, it has been repaired in 3.22.2.  https://github.com/flutter/flutter/issues/148867#issuecomment-2155510864
 @danagbemava-nc @ignicubeLaeeq

**Comment by ignicubeLaeeq** [comment]
i have confirmed, is has been solved in 3.22.2.

**Comment by darshankawar** [comment]
Closing as resolved based on above comment.
